packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,206 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { Tracker } from 'meteor/tracker'
+import { EJSON } from 'meteor/ejson'
+import { useState, useEffect, useReducer, useRef, DependencyList, Reducer } from 'react'
+
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+type useSubscriptionRefs = {
+  subscription?: Meteor.SubscriptionHandle,
+  updateOnReady: boolean,
+  isReady: boolean,
+  params: {
+    name?: string,
+    args: any[]
+  }
+}
+
+const useSubscriptionClient = (name?: string, ...args: any[]): Meteor.SubscriptionHandle => {
+  const forceUpdate = useForceUpdate()
+
+  const refs: useSubscriptionRefs = useRef({
+    updateOnReady: false,
+    isReady: false,
+    params: {
+      name,
+      args
+    }
+  }).current
+
+  if (!EJSON.equals(refs.params, { name, args })) {
+    refs.updateOnReady = false
+    refs.isReady = false
+    refs.params = { name, args }
+  }
+
+  useEffect(() => {
+    const computation = Tracker.nonreactive(() => (
+      Tracker.autorun(() => {
+        const { name, args } = refs.params
+        if (!name) return
+
+        refs.subscription = Meteor.subscribe(name, ...args)
+
+        const isReady = refs.subscription.ready()
+        if (isReady !== refs.isReady) {
+          refs.isReady = isReady
+          if (refs.updateOnReady) {
+            forceUpdate()
+          }
+        }
+      })
+    ))
+
+    return () => {
+      computation.stop()
+      delete refs.subscription
+    }
+  }, [refs.params])
+
+  return {
+    stop () {
+      refs.subscription?.stop()
+    },
+    ready () {
```

Comment:
As we discussed I think that it's potentially confusing to users to return a faux subscription handle here, as for example, users would expect this to work, and find that it doesn't:

```
useTracker(() => { setReady(handle.ready()) }, [handle]);
```

I'd suggest returning an array containing loading state along with the real handle `[loading, handle]`.

Reply:
Changing the function name of `.ready()` to something else probably makes sense (maybe `.isLoading()`) but it should stay a function. There's a small "feature" here, where it won't trigger a rerender when the ready state changes, unless you actually call the `.ready()` function.

```js
// each is a function
const { isLoading, stop } = useSubsription(...);

if (isLoading()) {
  return <Loading />
} else {
  return <ListContainer />
}
```

Reply:
Is the only reason to not return a state value to avoid an extra render?

Reply:
Yes, I believe so. The extra render is usually unnecessary as the new data will already trigger updates. It's only necessary when you have a specific loading state that depends on it.

Reply:
I changed it to be `[loading(), reactiveSubscriptionHandle | undefined]`